tests/extras/datasets/spark/test_spark_dataset.py

```diff
@@ -25,7 +25,6 @@
 #
 # See the License for the specific language governing permissions and
 # limitations under the License.
```

Comment:
We've removed this boilerplate from all files, so make sure you update your branch to make this disappear. 💥

Reply:
> Thank you for creating this PR @lvijnck. Being able to read with a schema is a valuable addition to the dataset. I have some question around whether it is definitely always `json` or if we're coding ourselves into a corner by limiting it that way.

Hi @lorenabalan 👋🏻! Thank you for the review.

Exactly, had the same question. Maybe the schema load argument could be complex, where the user has to specify a `type` (which could be inferred from the file extension by default?).

Reply:
Discussed live, updated PR description.